sanic/request.py

```diff
@@ -200,21 +203,73 @@ def files(self):
 
         return self.parsed_files
 
-    @property
-    def args(self):
-        if self.parsed_args is None:
+    def get_args(
+        self,
+        keep_blank_values: bool = False,
+        strict_parsing: bool = False,
+        encoding: str = "utf-8",
+        errors: str = "replace",
+    ) -> RequestParameters:
+        if not self.parsed_args[
+            (keep_blank_values, strict_parsing, encoding, errors)
+        ]:
             if self.query_string:
-                self.parsed_args = RequestParameters(
-                    parse_qs(self.query_string)
+                self.parsed_args[
+                    (keep_blank_values, strict_parsing, encoding, errors)
+                ] = RequestParameters(
+                    parse_qs(
+                        qs=self.query_string,
+                        keep_blank_values=keep_blank_values,
+                        strict_parsing=strict_parsing,
+                        encoding=encoding,
+                        errors=errors,
+                    )
                 )
-            else:
-                self.parsed_args = RequestParameters()
-        return self.parsed_args
+
+        return self.parsed_args[
+            (keep_blank_values, strict_parsing, encoding, errors)
+        ]
+
+    args = property(get_args)
 
     @property
-    def raw_args(self):
+    def raw_args(self) -> dict:
+        if self.app.debug:  # pragma: no cover
+            warnings.simplefilter("default")
+        warnings.warn(
+            "Use of raw_args will be deprecated in "
+            "the future versions. Please use args or query_args "
+            "properties instead",
+            DeprecationWarning,
+        )
         return {k: v[0] for k, v in self.args.items()}
 
+    def get_query_args(
+        self,
+        keep_blank_values: bool = False,
```

Comment:
The Original suggestion was the add the `keep_blank_values` to the `parse_qsl` directly and not impact any of the existing behaviors. This would exist only in the new method/property being added and not modify any of the existing behaviors.

Reply:
It is not normal that 1 parser `args` will return values with `keep_blank_values=False` and new one will use `keep_blank_values=True`, so I should have changed both.
I think that it is even worst to just change the default behavior.
And now for those who already use `args` - nothing has changed. 
If they need `keep_blank_values=True` - they will just use `get_args(keep_blank_values=True)`.
If somebody will need to change some other parameters of `parse_qs` / `parse_qsl` now it is very easy.

Reply:
@Amanit  Works for me 👍  Thanks for the clarification.   Can you please make sure that the API method is documented too with proper documentation? I know there is a valid example, having APIs documented helps too.

Reply:
Added docstrings for these methods.